cpp/include/raft/neighbors/detail/ivf_pq_build.cuh

```diff
@@ -442,15 +442,16 @@ void train_per_subset(raft::resources const& handle,
                  stream);
 
     // train PQ codebook for this subspace
-    auto sub_trainset_view =
-      raft::make_device_matrix_view<const float, IdxT>(sub_trainset.data(), n_rows, index.pq_len());
-    auto centers_tmp_view = raft::make_device_matrix_view<float, IdxT>(
+    auto sub_trainset_view = raft::make_device_matrix_view<const float, int64_t>(
```

Comment:
Do I understand it right, that all of the changes in this file are replacing `IdxT` with `int64_t`, and the purpose of this is to reduce the number of instantiations of `kmeans_balanced`?
This file is in `detail` namespace. Would it make sense to define something like `using internal_extents_t = int64_t` somewhere above in this file and use that one instead of `int64_t`? I hope this would make the intent clear, in case if someone needs to update the file in future and wonders where to use `IdxT` and where to use `int64_t`.

Reply:
Yes, this change only affects the mdpan extent type internally, and it is only made to reduce the number of template instantiations. Thanks for the good suggestion, I have made the change accordingly.